pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -490,6 +493,46 @@ monitorLoop:
 	}
 }
 
+func (l *LibvirtDomainManager) CancelVMIMigration(vmi *v1.VirtualMachineInstance) error {
+	if vmi.Status.MigrationState == nil || vmi.Status.MigrationState.Completed ||
+		vmi.Status.MigrationState.Failed {
+
+		l.setMigrationResult(vmi, false, "", v1.MigrationAbortFailed)
+		return fmt.Errorf("failed to cancel migration - vmi is not migrating")
+	}
+	l.asyncMigrationAbort(vmi)
+	return nil
+}
+
+func (l *LibvirtDomainManager) asyncMigrationAbort(vmi *v1.VirtualMachineInstance) {
+	go func(l *LibvirtDomainManager, vmi *v1.VirtualMachineInstance) {
```

Comment:
does this have to be in a go-routine?  what's blocking?

Reply:
AbortJob() is blocking

Reply:
ah, i see. thanks!